packages/types/src/modules-sdk/index.ts

```diff
@@ -227,14 +227,20 @@ export declare type ModuleJoinerRelationship = JoinerRelationship & {
 export type ModuleExports = {
   service: Constructor<any>
   loaders?: ModuleLoaderFunction[]
+  /**
+   * @deprecated property will be removed in future versions
+   */
   migrations?: any[]
+  /**
+   * @deprecated property will be removed in future versions
+   */
   models?: Constructor<any>[]
   runMigrations?(
-    options: LoaderOptions,
+    options: LoaderOptions<any>,
     moduleDeclaration?: InternalModuleDeclaration
   ): Promise<void>
   revertMigration?(
-    options: LoaderOptions,
+    options: LoaderOptions<any>,
     moduleDeclaration?: InternalModuleDeclaration
   ): Promise<void>
```

Comment:
we will need to update all the modules, they are all exporting it from the entry point, and none are proving these functions from the moduleDefinition. wdyt? and this (loadModuleMigrations) will change as well right from the load internals?

Reply:
good catch. updated them all